amodevices/hvs_controller/hvs_controller.py

```diff
@@ -0,0 +1,195 @@
+
+from .. import dev_generic
+from ..dev_exceptions import DeviceError
+
+logger = logging.getLogger(__name__)
+
+class hvsController(dev_generic.Device):
+
+    def __init__(self, device):
+        super().__init__(device)
+
+        self._monitors = PyDAQmx.Task("Voltage Monitors")
+        self._voltage_controls = PyDAQmx.Task("Voltage Controls")
+
+        self.timeout = device["Timeout"]
+
+        self._cache_interval = 0.5
+        self._last_reading_time = None
+        self._voltages = np.full(shape=(4,), fill_value=np.nan)
+        self._currents = np.full(shape=(2,), fill_value=np.nan)
+        self._write_voltages = np.zeros(4)
+
+        self._connected = False
+
+    def connect(self, adc_slot='cDAQ1Mod4', dac_slot='cDAQ1Mod2'):
+        """Setup the necessary 'connections' for the high voltage controller."""
+        try:
+            for monitor in self.device['Monitors']:
+                adc_port = str.encode(f"/{adc_slot}/AI{monitor['Input']}")
+                if monitor["Measurement"] == "Voltage":
+                    # Double check that this works? How does computer assign port?
+                    self._monitors.CreateAIVoltageChan(adc_port,
+                                                        b'',
+                                                        PyDAQmx.DAQmx_Val_RSE,
+                                                        PyDAQmx.float64(0),
+                                                        PyDAQmx.float64(monitor['Limit']),
+                                                        PyDAQmx.DAQmx_Val_Volts,
+                                                        None)
+                elif monitor["Measurement"] == "Current":
+                    self._monitors.CreateAIVoltageChan(adc_port,
+                                                        b'',
+                                                        PyDAQmx.DAQmx_Val_RSE,
+                                                        PyDAQmx.float64(0),
+                                                        PyDAQmx.float64(monitor['Limit']),
+                                                        PyDAQmx.DAQmx_Val_Volts,
+                                                        None)
+                else:
+                    raise DeviceError(
+                        f"{monitor['Measurement']} is not a known measurement type!")
+            for control in self.device['Controls']:
+                dac_port = str.encode(f"/{dac_slot}/AO{control['Output']}")
+                self._voltage_controls.CreateAOVoltageChan(dac_port,
+                                                           b'',
+                                                           PyDAQmx.float64(0),
+                                                           PyDAQmx.float64(control['Limit']),
+                                                           PyDAQmx.DAQmx_Val_Volts,
+                                                           None)
+            self._connected = True
```

Comment:
You should start the tasks here and then just read/write values below. There should then also be a `close` (or similar) method that stops the tasks. Starting and stopping a task takes a long time.